src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -247,44 +247,64 @@ namespace AppInstaller::CLI::Workflow
         {
-            InstallerTypeComparator(std::vector<InstallerTypeEnum> requirement) :
-                details::ComparisonField("Installer Type"), m_requirement(std::move(requirement))
+            InstallerTypeComparator(std::vector<InstallerTypeEnum> preference, std::vector<InstallerTypeEnum> requirement) :
+                details::ComparisonField("Installer Type"), m_preference(std::move(preference)), m_requirement(std::move(requirement))
             {
+                m_preferenceAsString = Utility::ConvertContainerToString(m_preference, InstallerTypeToString);
                 m_requirementAsString = Utility::ConvertContainerToString(m_requirement, InstallerTypeToString);
                 AICLI_LOG(CLI, Verbose,
-                    << "InstallerType Comparator created with Required InstallerTypes: " << m_requirementAsString);
+                    << "InstallerType Comparator created with Required InstallerTypes: " << m_requirementAsString
+                    << " , Preferred InstallerTypes: " << m_preferenceAsString);
             }
 
             static std::unique_ptr<InstallerTypeComparator> Create(const Execution::Args& args)
             {
+                std::vector<InstallerTypeEnum> preference;
                 std::vector<InstallerTypeEnum> requirement;
 
                 if (args.Contains(Execution::Args::Type::InstallerType))
                 {
                     requirement.emplace_back(Manifest::ConvertToInstallerTypeEnum(std::string(args.GetArg(Execution::Args::Type::InstallerType))));
                 }
+                else
+                {
+                    preference = Settings::User().Get<Settings::Setting::InstallerTypePreference>();
+                    requirement = Settings::User().Get<Settings::Setting::InstallerTypeRequirement>();
+                }
 
-                if (!requirement.empty())
+                if (!preference.empty() || !requirement.empty())
                 {
-                    return std::make_unique<InstallerTypeComparator>(requirement);
+                    return std::make_unique<InstallerTypeComparator>(preference, requirement);
                 }
                 else
                 {
                     return {};
                 }
             }
 
+            std::string ExplainInapplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                std::string result = "InstallerType does not match required type: ";
+                result += InstallerTypeToString(installer.EffectiveInstallerType());
+                result += "Required InstallerTypes: ";
+                result += m_requirementAsString;
+                return result;
+            }
+
+            bool ContainsInstallerType(const std::vector<InstallerTypeEnum>& selection, InstallerTypeEnum installerType)
+            {
+                return std::find(selection.begin(), selection.end(), installerType) != selection.end();

```

Comment:
Also, thinking more about this, I wonder if it would make sense to have a templated utility function for checking if an object is contained within a container. 

Although, considering that this PR is required for 1.6, it might make sense not to create a utility function.

-----

I'm thinking something like this might work, but I'm not certain

```c++
template <template<typename, typename> typename Container, typename Allocator, typename Value>
bool ContainsObject(Container<Value, Allocator> container, Value value) {
  return std::find(container.begin(), container.end(), value) != container.end();
}
```